uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1053,33 +1063,119 @@ impl APIConverter<Field> for weedle::dictionary::DictionaryMember<'_> {
         if let Type::Object(_) = type_ {
             bail!("Objects cannot currently appear in record fields");
         }
+        let default = match self.default {
+            None => None,
+            Some(v) => Some(v.value.convert(ci, &type_)?),
+        };
         Ok(Field {
             name: self.identifier.0.to_string(),
             type_,
             required: self.required.is_some(),
-            default: match self.default {
-                None => None,
-                Some(v) => Some(v.value.convert(ci)?),
-            },
+            default,
         })
     }
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, Hash)]
+pub enum Radix {
+    Decimal = 10,
+    Octal = 8,
+    Hexadecimal = 16,
+}
```

Comment:
Is preserving the radix necessary for correctness in some cases, or is it entirely about readability of the generated code? (I'm not opposed to doing it for readability reasons, just want to make sure I understand).

Reply:
Preserving the radix is only for readabilty/documentation for the generated interfaces.

There are cases when we can't preserve radix (octal numbers, anyone?), or our choice of representation mean that it's not possible (negative hex and octal numbers); in both these cases we use decimal because correctness beats readability.

Reply:
Added a comment about radices.